dspace-api/src/main/java/org/dspace/app/bulkedit/MetadataExport.java

```diff
@@ -7,272 +7,101 @@
-     */
-    private static void printHelp(Options options, int exitCode) {
-        // print the help message
-        HelpFormatter myhelp = new HelpFormatter();
-        myhelp.printHelp("MetadataExport\n", options);
-        System.out.println("\nfull export: metadataexport -f filename");
-        System.out.println("partial export: metadataexport -i handle -f filename");
-        System.exit(exitCode);
-    }
-
-    /**
-     * main method to run the metadata exporter
-     *
-     * @param argv the command line arguments given
-     * @throws Exception if error occurs
-     */
-    public static void main(String[] argv) throws Exception {
-        // Create an options object and populate it
-        CommandLineParser parser = new PosixParser();
-
+    private Options constructOptions() {
         Options options = new Options();
 
         options.addOption("i", "id", true, "ID or handle of thing to export (item, collection, or community)");
+        options.getOption("i").setType(String.class);
         options.addOption("f", "file", true, "destination where you want file written");
+        options.getOption("f").setType(OutputStream.class);
+        options.getOption("f").setRequired(true);
         options.addOption("a", "all", false,
                           "include all metadata fields that are not normally changed (e.g. provenance)");
+        options.getOption("a").setType(boolean.class);
         options.addOption("h", "help", false, "help");
+        options.getOption("h").setType(boolean.class);
 
-        CommandLine line = null;
-
-        try {
-            line = parser.parse(options, argv);
-        } catch (ParseException pe) {
-            System.err.println("Error with commands.");
-            printHelp(options, 1);
-            System.exit(0);
-        }
 
-        if (line.hasOption('h')) {
-            printHelp(options, 0);
-        }
+        return options;
+    }
 
-        // Check a filename is given
-        if (!line.hasOption('f')) {
-            System.err.println("Required parameter -f missing!");
-            printHelp(options, 1);
+    public void internalRun() throws Exception {
+        if (help) {
+            handler.logInfo("\nfull export: metadata-export -f filename");
+            handler.logInfo("partial export: metadata-export -i handle -f filename");
+            printHelp();
+            return;
         }
-        String filename = line.getOptionValue('f');
 
-        // Create a context
-        Context c = new Context(Context.Mode.READ_ONLY);
-        c.turnOffAuthorisationSystem();
-
-        // The things we'll export
-        Iterator<Item> toExport = null;
-        MetadataExport exporter = null;
+        DSpaceCSV dSpaceCSV = metadataDSpaceCsvExportService
+            .handleExport(context, exportAllItems, exportAllMetadata, handle,
+                          handler);
+        handler.writeFilestream(context, filename, dSpaceCSV.getInputStream(), "exportCSV");
```

Comment:
Since this "type" of "exportCSV" is going to end up saved into a metadata field (`process.type`), should we consider using a constant for the value (maybe in the Process class or similar?).  It seems odd to hardcode it if the value means something to the Process.

Reply:
Again, please resolve prior feedback here.